src/ruby/puppetserver-lib/puppet/server/http_client.rb

```diff
@@ -96,7 +94,15 @@ def create_common_request_options(url, headers, options)
 
     # Ensure multiple requests are not made on the same connection
     headers["Connection"] = "close"
-    request_options = RequestOptions.new(build_url(url))
+
+    # These defaults match what the agent attaches to its HTTP requests
+    headers["X-Puppet-Version"] ||= Puppet.version
```

Comment:
Do we want this to be the Puppet Server version instead? Not sure if that's readily available from here, but we _might_ be able to make that happen.

Reply:
IMO `X-*-Version` is mostly useful to include in HTTP responses from the server so that the HTTP client can alter it's behavior, such as downgrade to an older protocol. It's less useful for HTTP clients to send in requests, because generally the client can negotiate by sending `Accept` header and query parameters. In puppet we first added it for the HTTP server case, but then also added it for the HTTP client case. I'd be fine leaving this out. But if you wanted to add it, I think adding `X-PuppetServer-Version` with the puppetserver version would be better?

Reply:
Okay. I think I'll leave it out for now. We can always add it later if we need it for something.

Reply:
Removed.